titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/AmazonStsAsyncProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.ext.aws;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceAsync;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceAsyncClientBuilder;
+
+@Singleton
+public class AmazonStsAsyncProvider implements Provider<AWSSecurityTokenServiceAsync> {
+
+    private final AWSSecurityTokenServiceAsync amazonStsAsync;
+
+    @Inject
+    public AmazonStsAsyncProvider(AwsConfiguration configuration, AWSCredentialsProvider credentialProvider) {
+        String region = configuration.getRegion().trim().toLowerCase();
+        this.amazonStsAsync = AWSSecurityTokenServiceAsyncClientBuilder.standard()
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("sts." + region + ".amazonaws.com", region))
```

Comment:
Can you just set a `.withRegion()` and have the endpoint automatically built? Or do you actually need to set the endpoint URL?

Reply:
If I remember correctly, in the past we had to do both. I am following this pattern to keep it consistent with other providers.